### PR TITLE
Story/stc 11

### PR DESCRIPTION
**Pull request by JKFerland**
I see able to merge, so I believe this is the correct request to merge STC-11 into develop.

### CODE REVIEW

File: .project [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<projectDescription>
+	<name>scriptoocloud</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+	</buildSpec>
+	<natures>
+	</natures>
+</projectDescription>
```
Comment (jdamerow) [inline .project:11]: please don't commit this file

File: scriptoocloud/pom.xml [hunk 2]
```diff
@@ -19,7 +19,7 @@
 		<admin.password>$2a$04$oQo44vqcDIFRoYKiAXoNheurzkwX9dcNmowvTX/hsWuBMwijqn44i</admin.password>
 
 		<db.driver>com.mysql.jdbc.Driver</db.driver>
-		<db.database.url>jdbc-url-to-db</db.database.url>
+		<db.database.url>db-url</db.database.url>
```
Comment (jdamerow) [inline scriptoocloud/pom.xml:22]: is there a reason you changed this?

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        	// long is a primitive data type, so we must convert to an object data type Long to check for null. 
+        	Long checkId = id;
+        	if(checkId == null) {
+        		throw new nullIDException("Project does not exist.");
+        	} else {
+        		projectRepo.deleteById(id);
+        	}
+        } catch (nullIDException err) {
+        	logger.error("Project does not exist.");
+        }
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java:62]: couple of things here:
- why do you assign `Long checkId = id`?
- there is no point in throwing an exception just to catch it right away. Throwing exceptions is to notify upper layers that something went wrong that can't be handled.
- Classes start with an uppercase letter (`nullIDException`)
- if you would catch an exception then it needs to be properly handled. Right now you're swallowing the exception. all that one would see in the logs would be "Project does not exist." No Stacktrace, no id, nothing. It wouldn't be useful at all to track down what went wrong.

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    }
+    
+    private Exception deleteNullIdException() {
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java:65]: what does this do?

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    */
+   @Override
+    public List<Project> findAll() {
+        Iterable<ProjectImpl> projects = projectRepo.findAll();
+        List<Project> results = new ArrayList<>();
+        projects.forEach(e -> results.add((Project) e));
+        return results;
+    }
+    
+}
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java:80]: Indentation is all over the place in this file.

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/nullIDException.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+public class nullIDException extends Exception {
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/nullIDException.java:3]: Classes start with an uppercase letter. Eclipse should have given you a warning when you created this class.

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/AddProjectController.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+@Controller
+public class AddProjectController {
+    
+    @Autowired
+    private ProjectManager projectManager;
+    
+    private final IUserManager userManager;
+
+    @Autowired
+    public AddProjectController(IUserManager userManager) {
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/AddProjectController.java:34]: Be consistent, either autowire all fields or autowire the constructor but try not to mix them (put autowired annotation on IUserManager declaration or pass ProjectManager into constructor (and remove autowired annotation on the variable declaration).

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/AddProjectController.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        this.userManager = userManager;
+    }
+
+    @RequestMapping(value = "/auth/add", method=RequestMethod.GET)
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/AddProjectController.java:38]: add what? The path should be specific to the elements you're working with.

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/AddProjectController.java [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    public String get(Model model) {
+    	model.addAttribute("project", new ProjectImpl());
+        return "/auth/add";
+    }
+    
+    @RequestMapping(value = "/auth/add", method=RequestMethod.POST)
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/AddProjectController.java:44]: see above

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/AddProjectController.java [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+            model.addAttribute("project", projectImpl);
+        }
+        
+        String username = principal.getName();
+        IUser user = userManager.findByUsername(username);
+        
+        Project project = projectManager.createProject(projectImpl.getName(), projectImpl.getDescription(), user);
+        return "redirect:/auth/projects";
+    }
+}
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/AddProjectController.java:56]: indentation

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/ProjectsController.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+    @Autowired
+    private ProjectManager projectManager;
+
+    @RequestMapping("/auth/projects")
+    public String listProjects(Model model) {
+    	model.addAttribute("projects", projectManager.findAll() );
+    	return "/auth/projects";
+    }
+    
+}
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/ProjectsController.java:42]: indentation

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+        return "/auth/projects/remove";
+    }              
+    
+    @RequestMapping(value = "/auth/projects/remove/{id}", method=RequestMethod.POST)
+    public String post(@PathVariable("id") int id,Model model) {
+        
+    	try {
+        	projectManager.deleteProject(id); 
+        	return "redirect:/auth/projects";
+        }catch (nullIDException err) {
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java:41]: this exception is never thrown by the delete method, is it?

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+    
+    @Autowired
+    private ProjectManager projectManager;
+    
+    @RequestMapping(value = "/auth/projects/remove", method=RequestMethod.GET)
+    public String get(Model model) {
+        return "/auth/projects/remove";
+    }              
+    
+    @RequestMapping(value = "/auth/projects/remove/{id}", method=RequestMethod.POST)
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java:35]: should be `/auth/projects/{id}/remove`

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+import edu.asu.diging.scriptoocloud.core.service.impl.nullIDException;
+
+
+@Controller
+public class RemoveProjectController {
+    
+    @Autowired
+    private ProjectManager projectManager;
+    
+    @RequestMapping(value = "/auth/projects/remove", method=RequestMethod.GET)
```
Comment (jdamerow) [inline scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java:30]: is this method ever used?

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: scriptoocloud/pom.xml (hunk 2 above)
Comment (JKFerland) [reply scriptoocloud/pom.xml:22]: I did not mean to commit this file/this change. I can not find my settings file to change this value within it (Does not appear where it should be the last time I checked for it). I had just put the database location here until I could find the settings file/re-setup the work environment

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java (hunk 14 above)
Comment (JKFerland) [reply scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java:30]: No, removed.

File: scriptoocloud/pom.xml (hunk 2 above)
Comment (jdamerow) [reply scriptoocloud/pom.xml:22]: Your settings file should be in `your home directory/.m2/settings.xml`.

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java (hunk 13 above)
Comment (JKFerland) [reply scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java:35]: Changed

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/nullIDException.java (hunk 6 above)
Comment (JKFerland) [reply scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/nullIDException.java:3]: I did not get a red or yellow error about this, but still an oversight on my end. Refactored, and also moved to /core/exceptions.

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java (hunk 4 above)
Comment (JKFerland) [reply scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java:65]: Unsure why I left this here, removed.

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java (hunk 3 above)
Comment (JKFerland) [reply scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/ProjectManagerImpl.java:62]: -checkID removed, had another idea in mind but is now removed. 
-Removed try/catch block in ProjectManagerImpl. I think my understanding here was not correct. Should now be caught appropriately in the RemoveProjectController. 
-Refactored nullIDException to NullIDException
-With correct catching of the block (if I fixed it correctly), should display "Project {id} does not exist".

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/ProjectsController.java (hunk 11 above)
Comment (JKFerland) [reply scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/ProjectsController.java:42]: Formatted with Eclipse, hope this fixes this reoccurring issue I am having.

File: scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java (hunk 12 above)
Comment (JKFerland) [reply scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/RemoveProjectController.java:41]: Should now be thrown by the delete method.

**Pull request closed without merging**